pkg/controllermanager/apis/config/v1alpha1/defaults.go

```diff
@@ -210,7 +209,7 @@ func SetDefaults_ClientConnectionConfiguration(obj *componentbaseconfigv1alpha1.
 // SetDefaults_LeaderElectionConfiguration sets defaults for the leader election of the Gardener controller manager.
 func SetDefaults_LeaderElectionConfiguration(obj *componentbaseconfigv1alpha1.LeaderElectionConfiguration) {
 	if obj.ResourceLock == "" {
-		obj.ResourceLock = resourcelock.LeasesResourceLock
+		obj.ResourceLock = "leases"
```

Comment:
This is not obvious, I would add a comment so that we don't reintroduce this dependency later again.
(same in the other files)